src/subs.jl

```diff
@@ -91,30 +91,48 @@ end
 
 evaluates a symbolless expression or returns a function
 """
-function lambdify(ex::Basic)
-    vars = free_symbols(ex)
+function lambdify(ex::Basic, vars=[])
     if length(vars) == 0
-        _lambdify(ex)
+        vars = free_symbols(ex)
+    end
+    body = walk_expression(ex)
+
+    if length(vars) == 0
+        # return a number
+        eval(body)
     else
-        _lambdify(ex, vars)
+        # return a function
+        _lambdify(body, vars)
     end
 end
-lambdify(ex::BasicType) = lambdify(Basic(ex))
+lambdify(ex::BasicType, vars=[]) = lambdify(Basic(ex), vars)
+
+function lambdify(m::AbstractArray{Basic, 2}, vars=[])
```

Comment:
This supports 2D array only atm.

Reply:
I think you can do it with a generated function to get the dimension parameter and then using Base.Cartesian. I'm not sure on the exact details of how to do it though.